packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,185 @@
+# Table Column Widths
+
+## Problem Statement
+
+We need to provide users the means for changing the widths of individual columns as needed. This should be able to be accomplished both through interactive means as well as via a programmatic API.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [#873 Programmatically resize column width](https://github.com/ni/nimble/issues/873)
+-   [#846 Interactively resize column width](https://github.com/ni/nimble/issues/846)
+-   [Table README](./README.md)
+-   [Table Design Doc](https://xd.adobe.com/view/5b476816-dad1-4671-b20a-efe796631c72-0e14/screen/d389dc1e-da4f-4a63-957b-f8b3cc9591b4/specs/)
+
+## Implementation / Design
+
+### Expected Behavior
+
+The behavior that has been prescribed for column sizing is as follows:
+
+-   Columns should be able to be configured to either maintain a fixed width, or grow proportionally with the table such as when the window resizes causing the table width to increase. Tables can consist of columns that are configured as a mixture of the two modes.
```

Comment:
Some positive feedback about this approach: we described it to Josh today and he was thrilled that we'd have fixed width columns to allow icon columns to waste less horizontal space.

Reply:
It's interesting that Josh was "thrilled" about this, given the fact that this is the way all the tables work today in SLE. Do you know if he was viewing this as an improvement over some other table(s) in our products today?

Reply:
His impression of how the table worked today was that there were some columns that should be fixed width but were not, resulting in them taking up more width than necessary. I took a quick scan in SLE and I couldn't find any examples of this. So it's possible the impression came from SLS or maybe it's mistaken/outdated.